### PR TITLE
Bash completion for runner 2392

### PR DESCRIPTION
**Pull request by mariaschett**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: scripts/generate_bash_completion_for_runner [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+echo "
+Wrote the bash completion file for runner to ${BASH_COMPLETION_FILE}
+Please source this file in your .bashrc, e.g., through
+  
+echo \"
+#activate runner auto-complete
+if [ -f $BASH_COMPLETION_FILE ]; then
+  source $BASH_COMPLETION_FILE
+fi\" >> ~/.bashrc
+"
```
Comment (conradgrobler) [inline scripts/generate_bash_completion_for_runner:19]: nit: missing line break an end of file.

**Review by conradgrobler — APPROVED** [review]
Nice, thanks.

File: scripts/generate_bash_completion_for_runner [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+
+readonly BASH_COMPLETION_DIR="$(realpath ".")"
+readonly BASH_COMPLETION_FILE=$BASH_COMPLETION_DIR/.runner_bash_completion 
+
+./scripts/runner completion --out-dir="$BASH_COMPLETION_DIR"
```
Comment (tiziano88) [inline scripts/generate_bash_completion_for_runner:7]: Is it ok to do all this work in bashrc? IIUC it is compiling and running runner every time a new bash shell is created. I thought you suggested another approach in the past, of checking in the completion file and just sourcing that, is that not an option any more?

File: Dockerfile [hunk 3]
```diff
@@ -319,3 +319,12 @@ ENV CARGO_INCREMENTAL false
 # uid of the host, therefore we need to first fix the uid before actually using the user. This is
 # done by /scripts/fix_docker_user_and_run .
 RUN useradd --shell=/bin/bash --create-home --user-group docker
+
+# To make the scripts available to call from everywhere.
+ENV PATH "/workspace/scripts:${PATH}" 
+
+# Add sourcing of runner_bash_completion file to .bashrc
+RUN echo -e "\n#activate runner auto-complete\nif [ -f /workspace/.runner_bash_completion ]; then\n  source /workspace/.runner_bash_completion \nfi" >> /home/docker/.bashrc
```
Comment (mariaschett) [inline Dockerfile:327]: Here we add the command to `.bashrc` to source the checked in `workspace/.runner_bash_completion` file.

Comment (tiziano88) [reply Dockerfile:327]: Ah my bad, github was hiding that diff because it was too large :grimacing: makes sense, thanks!

**Review by rbehjati — APPROVED** [review]
Nice. Thanks :)

File: scripts/runner [hunk 4]
```diff
@@ -5,5 +5,5 @@ readonly SCRIPTS_DIR="$(dirname "$0")"
 source "$SCRIPTS_DIR/common"
 
 sccache --show-stats
-cargo run --manifest-path=./runner/Cargo.toml -- "$@"
+cargo run --manifest-path="$SCRIPTS_DIR/../runner/Cargo.toml" -- "$@"
```
Comment (ipetr0v) [inline scripts/runner:8]:
```suggestion
cargo run --manifest-path="${SCRIPTS_DIR}/../runner/Cargo.toml" -- "$@"
```

File: scripts/generate_bash_completion_for_runner [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+
+echo "
+Wrote the bash completion file for runner to ${BASH_COMPLETION_FILE}
+Please source this file in your .bashrc, e.g., through
+  
+echo \"
+#activate runner auto-complete
+if [ -f $BASH_COMPLETION_FILE ]; then
```
Comment (ipetr0v) [inline scripts/generate_bash_completion_for_runner:16]:
```suggestion
if [ -f "${BASH_COMPLETION_FILE}" ]; then
```

File: scripts/generate_bash_completion_for_runner [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  source $BASH_COMPLETION_FILE
```
Comment (ipetr0v) [inline scripts/generate_bash_completion_for_runner:17]:
```suggestion
  source "${BASH_COMPLETION_FILE}"
```

**Review by ipetr0v — APPROVED** [review]

File: scripts/generate_bash_completion_for_runner (hunk 1 above)
Comment (mariaschett) [reply scripts/generate_bash_completion_for_runner]: fixed :)

**Pull request merged**